processmanager/types.go

```diff
@@ -138,15 +137,38 @@ func (m *Mapping) GetOnDiskFileIdentifier() util.OnDiskFileIdentifier {
 	}
 }
 
-// addressSpace represents the address space of a process. It maps the known start addresses
-// of executable mappings to the corresponding mappedFile information.
-type addressSpace map[libpf.Address]Mapping
-
 // processInfo contains information about the executable mappings
 // and Thread Specific Data of a process.
 type processInfo struct {
-	// executable mappings
-	mappings addressSpace
+	// executable mappings keyed by start address.
+	mappings map[libpf.Address]*Mapping
+	// executable mappings keyed by host file ID.
+	mappingsByFileID map[host.FileID]map[libpf.Address]*Mapping
```

Comment:
I added this new lookup map for allow retrieving mappings for file IDs in `O(1)`; otherwise we'd have to do an `O(n)` linear sweep through the `mappings` map above for every frame being converted in `ConvertTrace`.